.github/workflows/_check-user.yml

```diff
@@ -0,0 +1,72 @@
+name: Check whether the workflow owner can use ARC runners
+
+on:
+  workflow_call:
+    inputs:
+      user_name:
+        required: true
+        type: string
+        description: The name of the workflow owner.
+      curr_branch:
+        required: true
+        type: string
+        description: Current branch.
+
+    outputs:
+      workflow-type:
+        description: Type of runners to use
+        value: ${{ jobs.check-user.outputs.workflow-type }}
```

Comment:
I believe that it should output both the `workflow-type` and `runner`. So we can use it to easily migrate jobs from one place to another. Dynamically.

Makes more sense to do it in this way and maintain the flexibility to quickly migrate workflows in the future.

Reply:
I general, I agree with you - we can use this approach dynamically selecting a runner. We don't need this feature at the moment. Let's keep the first version light and then iteratively improve it.